packages/lib/src/composables/events.ts

```diff
@@ -30,47 +30,36 @@ const CHART_EVENTS = [
   'tooltip-opened',
   'tooltip-moved',
   'tooltip-closed',
-];
-
-interface EventBusInstance {
-  $on: (
-    event: string | Array<string>,
-    fn: (...args: Array<unknown>) => void
-  ) => void;
-  $emit: (event: string, ...args: Array<unknown>) => void;
-}
-
-interface EventBusInstanceConstructor {
-  new (): EventBusInstance;
-}
 
-let eventBus: EventBusInstance | null = null;
+  'group-mouseenter',
+  'group-mouseleave',
+];
 
 // Used to propagate events from the top-most component (needed for Vue 2)
 export const useEvents = (
   emit: (e: string, p: unknown) => void,
   chartID: string
 ) => {
-  async function initEventBus() {
-    if (__VUE_VERSION__ === 3) return null;
-
-    const Vue = (await import('vue')).default;
-    eventBus = new (Vue as unknown as EventBusInstanceConstructor)();
+  function emitInternalEvent(eventName: string, ...args: Array<unknown>) {
+    if (__VUE_VERSION__ === 2) {
+      document.dispatchEvent(
+        new CustomEvent(`${chartID}_${eventName}`, {
+          detail: [...args],
+        })
+      );
+    }
   }
 
-  async function busEmit(eventName: string, ...args: Array<unknown>) {
-    if (!eventBus) await initEventBus();
-
-    eventBus.$emit(`${chartID}_${eventName}`, ...args);
-  }
-
-  async function busListen(
+  function listenInternalEvent(
     eventName: string,
     listener: (...args: Array<unknown>) => void
   ) {
-    if (!eventBus) await initEventBus();
-
-    eventBus.$on(`${chartID}_${eventName}`, listener);
+    if (__VUE_VERSION__ === 2) {
+      document.addEventListener(
+        `${chartID}_${eventName}`,
+        (e: CustomEventInit) => listener(e.detail[0])
```

Comment:
Should pass all `args` and not just the first
```suggestion
        (e: CustomEventInit) => listener(...e.detail)
```

Reply:
I don't get it: let's say we do `emitInternalEvent('evt-name', 123, 456)`, then `e.detail` is `[123, 456]`, right?
By doing `e.detail[0]` we're only passing `123`

Reply:
Aaah, I see your point!
There's a catch here.

So, the thing is - we're creating `emitInternalEvent` this way:

```
 document.dispatchEvent(
        new CustomEvent(`${chartID}_${eventName}`, {
          detail: [...args],
        })
 );
```

So, e.detail is `[[123, 456]]`, as a result `e.detail[0]` would be `[123, 456]`

Had to dispatch it as an array of array to pass the type check. Otherwise CustomEventInit type check fails 😅

Reply:
But you're not sending an array of arrays, you are sending just one array which is how it should be - problem is going to index 0. Check here:

![image](https://github.com/Adyen/lume/assets/31653199/603b92ad-26a2-451c-82cc-34d94cb2cfd1)

I added a `console.log` of what `listenInternalEvent` is getting in `e.detail` + dispatched an extra argument - you can see that it is a normal 1 dimension array, which means `e.detail[0]` will be just the number and will not pass "something else"

Reply:
Indeed indeed!
Though I was spreading and boxing (`[...args]`) - it was still an array (not array of arrays) - lol 🤦 

Agreed, will update :)